src/editor/map/map_context.cpp

```diff
@@ -777,7 +777,46 @@ config map_context::to_config()
 	return scen;
 }
 
-bool map_context::save_scenario()
+void map_context::save_schedule()
+{
+	/* Create schedule config */
+	config schedule;
+	config& editor_times = schedule.add_child("editor_times");
+	config times = tod_manager_->to_config();
+	for(const config& time : times.child_range("time")) {
+		config& t = editor_times.add_child("time");
+		t.append(time);
+	}
+
+	/* Write to file */
+	try {
+		std::stringstream wml_stream;
+		wml_stream
+			<< "# This file was generated using the scenario editor.\n"
```

Comment:
Too much copied from `save_scenario`?

Reply:
Yes, I'm following how `save_scenario` is written to a file. How can I improve or simplify this?

Reply:
Please ignore my comment - somehow I missed that this was writing a separate file, assumed that the schedule was being written to the same file as the scenario, and thus that this would duplicate the warning that was already in the same file.

Reply:
Okay.
A couple of questions.
1. Are the braces around the config writer necessary.
2. Is the if block really necessary? since wml_stream can never actually be empty due to the "#Editor" stuff.
`if(!wml_stream.str().empty()) {
			filesystem::write_file(filesystem::get_current_editor_dir(addon_id_) + "/utils/schedule.cfg", wml_stream.str());
}`

Reply:
1. It looks like it's technically unnecessary since you're writing without compression, but I think it's probably better to include them if you want to use `config_writer`. There is also the possibility of directly calling the global `write` function though, in which case it wouldn't make any sense to have braces.
2. Yeah, you seem to be correct that the stream will never be empty. However, I wonder if the intent of that check is more about whether the config writer wrote anything? After all, writing a file that just contains the warning and an empty `#ifdef` block isn't exactly useful.

Reply:
1. Okay, they can stay.
2. Maybe the check can be modified, so that it doesn't writes anything if there's no schedule? That is, the warning and empty `#ifdef`won't be written if there's no `[time]` block.

Reply:
Probably a issue should be created as to why the "_" is not being written to the file.

Reply:
Did my comments from the chat not help? With the current code I think it's correct that the _ is not being written.

Reply:
> Did my comments from the chat not help? With the current code I think it's correct that the _ is not being written.

No, you explained it well. But the problem is the `config_writer` and the related files are quite complicated for me to understand yet, so I haven't managed to figure out exactly what is stopping the '_' from being written.

Reply:
> so I haven't managed to figure out exactly what is stopping the '_' from being written.
Finally found it out. The right hand sides of assignments of keys (`cfg["name"] = name`) needs to be changed to (`cfg["name"] = t_string(name, textdomain)`). This problem exists in `map_context.cpp` and any other places where `config` objects are being created.

Reply:
I think I said the same thing in chat, didn't I? What are you using for the textdomain though?

Reply:
> I think I said the same thing in chat, didn't I?
Yes, took me a while to understand that `config_writer` isn't causing the problem and it's located elsewhere. Got confused.